### PR TITLE
Chat/integration

### PR DESCRIPTION
**Pull request by cjmconie**

### CODE REVIEW

File: BlockV/Classes/BLOCKv+Request.swift [hunk 1]
```diff
@@ -25,7 +25,9 @@ extension BLOCKv {
     ///               Only the properties to be registered should be set.
     ///   - completion: The completion handler to call when the request is completed.
     ///                 This handler is executed on the main queue.
-    public static func register(withUserToken token: String, type: UserTokenType, userInfo: UserInfo? = nil,
```
Comment (LordCheddar) [inline BlockV/Classes/BLOCKv+Request.swift]: Can you please do unrelated formatting changes on a separate branch in the future.

File: BlockV/Classes/BLOCKv+Request.swift [hunk 2]
```diff
@@ -892,4 +913,78 @@ extension BLOCKv {
...
+                completion(threadListModel, nil)
+            }
+            
+        }
+        
+    }
+    
+    /// Fetches the activity messages, for a specified thread id, after the specified cursor.
+    ///
+    /// Only messages involving the current user are returned.
```
Comment (LordCheddar) [inline BlockV/Classes/BLOCKv+Request.swift]: Should this not be "Only messages from a thread involving the current user are returned."

File: BlockV/Classes/Network/Models/Activity/ThreadModel.swift [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        
+    }
+    
+    /// Unique identifier of the message thread.
+    ///
+    /// The id is a compound name <user_a>:<user_b>
+    public let id: String
+    /// HACK - server returns the date as a Double.
+    private let _whenModified: Double
```
Comment (LordCheddar) [inline BlockV/Classes/Network/Models/Activity/ThreadModel.swift]: Thought we were leaving this as a double, this 'hack' will require to be across all sdks otherwise

File: BlockV/Classes/Network/Models/Geo Group/GeoGroupModel.swift [hunk 4]
```diff
@@ -39,7 +39,7 @@ public struct GeoGroupModel: Equatable {
     /// Geo hash. Useful for URLs etc.
     public let geoHash: String
     /// Coordinate of the group.
-    public let location: CLLocationCoordinate2D
+    public let coordinate: CLLocationCoordinate2D
```
Comment (LordCheddar) [inline BlockV/Classes/Network/Models/Geo Group/GeoGroupModel.swift]: Please do unrelated changes in a separate pr

Comment (cjmconie) [reply BlockV/Classes/Network/Models/Geo Group/GeoGroupModel.swift]: Noted

File: BlockV/Classes/BLOCKv+Request.swift (hunk 1 above)
Comment (cjmconie) [reply BlockV/Classes/BLOCKv+Request.swift]: Noted.

File: BlockV/Classes/BLOCKv+Request.swift (hunk 2 above)
Comment (cjmconie) [reply BlockV/Classes/BLOCKv+Request.swift]: I have removed that comment as it was duplicating information from the comment above.

File: BlockV/Classes/Network/Models/Activity/ThreadModel.swift (hunk 3 above)
Comment (cjmconie) [reply BlockV/Classes/Network/Models/Activity/ThreadModel.swift]: As agreed, I will leave it as a `Date` type.
